src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorNetworkIT.java

```diff
@@ -0,0 +1,184 @@
+import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConfig.LINGER_MS_CONFIG;
+import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConfig.MAX_IN_FLIGHT_REQUESTS_CONFIG;
+import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConfig.MAX_RETRIES_CONFIG;
+import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConfig.READ_TIMEOUT_MS_CONFIG;
+import static io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConfig.RETRY_BACKOFF_MS_CONFIG;
+import static org.apache.kafka.connect.json.JsonConverterConfig.SCHEMAS_ENABLE_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.SinkConnectorConfig.TOPICS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class ElasticsearchConnectorNetworkIT extends BaseConnectorIT {
+
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(options().dynamicPort(), false);
+
+  protected static final int NUM_RECORDS = 5;
+  protected static final int TASKS_MAX = 1;
+
+  protected static final String CONNECTOR_NAME = "es-connector";
+  protected static final String TOPIC = "test";
+  protected Map<String, String> props;
+
+  @Before
+  public void setup() {
+    startConnect();
+    connect.kafka().createTopic(TOPIC);
+    props = createProps();
+
+    stubFor(any(anyUrl()).atPriority(10).willReturn(ok()));
+  }
+
+  @After
+  public void cleanup() {
+    stopConnect();
+  }
+
+  @Test
+  public void testReadTimeout() throws Exception {
+    wireMockRule.stubFor(post(urlPathEqualTo("/_bulk"))
+            .willReturn(ok().withFixedDelay(2_000)));
+
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, TASKS_MAX);
+    writeRecords(NUM_RECORDS);
+
+    // Connector should fail since the request takes longer than request timeout
+    await().atMost(Duration.ofMinutes(1)).untilAsserted(() ->
+            assertThat(connect.connectorStatus(CONNECTOR_NAME).tasks().get(0).state())
+                    .isEqualTo("FAILED"));
+
+    assertThat(connect.connectorStatus(CONNECTOR_NAME).tasks().get(0).trace())
+            .contains("Failed to execute bulk request due to 'java.net.SocketTimeoutException: " +
+                    "1,000 milliseconds timeout on connection")
+            .contains("after 3 attempt(s)");
+
+    // 1 + 2 retries
+    verify(3, postRequestedFor(urlPathEqualTo("/_bulk")));
+  }
+
+  @Test
+  public void testTooManyRequests() throws Exception {
```

Comment:
we should also add a new test for a scenario where the request to Elastic are taking a long time and the new ExecutorService has more requests than maxInFlightRequests